File: 240204/키를 기준으로 정렬/sort-by-height.java
import java.io.*;
import java.util.*;

class Student {
    String name;
    int height;
    int weight;
    public Student(String name, int height, int weight) {
        this.name = name;
        this.height = height;
        this.weight = weight;
    }
    public void print() {
        System.out.printf("%s %d %d\n", this.name, this.height, this.weight);
    }
}

public class Main {
    public static int inputLength;
    public static List<Student> students = new ArrayList<>();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        inputLength = Integer.parseInt(br.readLine());
        for (int i = 0; i < inputLength; i++) {
            String[] input = br.readLine().split(" ");
            String name = input[0];
            int height = Integer.parseInt(input[1]);
            int weight = Integer.parseInt(input[2]);
            students.add(new Student(name, height, weight));
        }

        Collections.sort(students, (a, b) -> a.height - b.height);
        for (Student student : students) {
            student.print();
        }
    }
}

File: 240317/키를 기준으로 정렬/sort-by-height.py
class Person:
    def __init__(self, name, height, weight):
        self.name = name
        self.height = height
        self.weight = weight

n = int(input())
persons = [Person(name, int(height), int(weight)) for _ in range(n) for name, height, weight in [input().split()]]

persons.sort(key=lambda ele: ele.height)
for ele in persons:
    print(ele.name, ele.height, ele.weight)